tests/unit/src/commands/decryptDeployKey.ts

```diff
@@ -0,0 +1,133 @@
+import * as registerSuite from 'intern!object';
+import * as assert from 'intern/chai!assert';
+import loadModule, { cleanupModuleMocks } from '../../../_support/loadModule';
+import { stub, SinonStub } from 'sinon';
+
+let decryptDeployKey: any;
+let decryptDataObj: any;
+let decryptDataStub: SinonStub;
+let encryptedKeyFileStub: SinonStub;
+let keyFileStub: SinonStub;
+let existsSyncStub: SinonStub;
+let createWriteStreamStub: SinonStub;
+let createReadStreamStub: SinonStub;
+
+registerSuite({
+	name: 'commands/decryptDeployKey',
+
+	before() {
+		decryptDataObj = {
+			on: stub(),
+			pipe: stub()
+		};
+		decryptDataStub = stub();
+		encryptedKeyFileStub = stub();
+		keyFileStub = stub();
+		existsSyncStub = stub();
+		createWriteStreamStub = stub();
+		createReadStreamStub = stub();
+	},
+
+	after() {
+		cleanupModuleMocks();
+	},
+
+	beforeEach() {
+		decryptDataObj.on.returns(decryptDataObj);
+		decryptDataObj.pipe.returns(decryptDataObj);
+
+		decryptDeployKey = loadModule('src/commands/decryptDeployKey', {
+			'../util/crypto': {
+				decryptData: decryptDataStub.returns(decryptDataObj)
+			},
+			'../util/environment': {
+				decryptKeyName: 'decryptKeyName',
+				decryptIvName: 'decryptIvName',
+				encryptedKeyFile: encryptedKeyFileStub.returns('encryptedKeyFile'),
+				keyFile: keyFileStub.returns('keyFile')
+			},
+			'fs': {
+				existsSync: existsSyncStub,
+				createWriteStream: createWriteStreamStub.returns('writeStream'),
+				createReadStream: createReadStreamStub.returns('readStream')
+			}
+		});
+	},
+
+	afterEach() {
+		decryptDataStub.reset();
+		encryptedKeyFileStub.reset();
+		keyFileStub.reset();
+		existsSyncStub.reset();
+		createWriteStreamStub.reset();
+		createReadStreamStub.reset();
+		decryptDataObj.on.reset();
+		decryptDataObj.pipe.reset();
+	},
+
+	'decryptDeployKey': (() => {
+		return {
+			async 'arguments passed in explicitly'() {
+				ensureDecryptionResolves();
+
+				const deployKeyDecrypted = await assertDecryptDeployKey('encrypted.file', 'decrypt key', 'decrypt iv', 'decrypted.file');
+
+				assert.isTrue(deployKeyDecrypted);
```

Comment:
there's a lot of assertions here. Can we group them w/ their branches in code and add comments as to the condition that they're testing